File: TechnicalInterview/code/Stack.java
// Definition for a Task element in a Queue
class Task {
	private Object task;        // the task to perform
	private Task   next = null; // the next task in the queue

	// constructor, task represents the task to be performed
	public Task( Object task ) {
		this.task = task;
	}
	
	// Get the next task this element is chained to
	public Task Next() {
		return next;
	}
	
	// Set the next task this element is chained to
	public void Next( Task task ) {
		next = task;
	}
	
	// Action to take when task is processed
	public void Action() {
		System.out.println( task );
	}
}

// Definition for a Stack
public class Stack {
	private Task top;	// top of the stack
	
	// Check if the Stack is empty
	private boolean Empty() {
		if ( top == null )
			return true;
		return false;
	}
	
	// Push the task to the top of the stack
	public void Push( Task task ) {
		// Set the task next pointer to the current top
		task.Next( top );
		// set the top to this task
		top = task;
	}
	
	// Pop the task from the top of the stack
	public void Pop() {
		// Stack is Empty
		if ( Empty() )
			return;
			
		// Perform the action for the task
		top.Action();
		
		// Move the top to the current top's next pointer.
		top = top.Next();
	}

	public static void main(String[] args) {
		System.out.println("Process tasks in order: A, B and C" );
		Stack stack = new Stack();
		stack.Push( new Task( "C" ) );
		stack.Push( new Task( "B" ) );
		stack.Push( new Task( "A" ) );
		stack.Pop();
		stack.Pop();
		stack.Pop();
	}
}

File: TechnicalInterview/code/Stack.py
# Definition for a Task element in a Stack
class Task:
	task = None  # the task to perform
	next = None  # the next task in the stack

	# constructor, task represents the task to be performed
	def __init__( self, task ):
		self.task = task
	
	# Get the next task this element is chained to
	def GetNext(self):
		return self.next
	
	# Set the next task this element is chained to
	def Next( self, task ):
		self.next = task;
	
	# Action to take when task is processed
	def Action( self ):
		print( self.task )

# Definition for a Stack
class Stack:
	top = None   # top of the stack
	
	# Check if the Stack is empty
	def Empty(self):
		if self.top == None:
			return True
		return False
	
	# Push the task to the top of the stack
	def Push( self, task ):
		# Set the task next pointer to the current top
		task.Next( self.top )
		# Set the top to this task
		self.top = task
	
	# Pop the task from the top of the stack
	def Pop( self ):
		# Stack is Empty
		if self.Empty():
			return

		# Perform the action for the task
		self.top.Action()
		
		# Move the top to the current top's next pointer.
		self.top = self.top.GetNext()

print("Process tasks in order: A, B and C" )
stack = Stack()
stack.Push( Task( "C" ) )
stack.Push( Task( "B" ) )
stack.Push( Task( "A" ) )
stack.Pop()
stack.Pop()
stack.Pop()